src/Tribe/Tickets.php

```diff
@@ -3947,7 +3947,37 @@ public static function get_tickets_in_cart_for_provider( $provider = null ) {
 		 * @return string The generated security code.
 		 */
 		public function generate_security_code( $attendee_id ) {
-			return substr( md5( wp_rand() . '_' . $attendee_id ), 0, 10 );
+			$code = substr( md5( wp_rand() . '_' . $attendee_id ), 0, 10 );
+
+			/**
+			 * Filter the generated security code for tickets.
+			 *
+			 * @since 5.2.2
+			 *
+			 * @param string $code Generated secujrity code for attendee Ticket.
+			 * @param string $attendee_id ID of attendee.
+			 */
+			return apply_filters( 'tribe_tickets_attendee_security_code', $code, $attendee_id );
+		}
+
+		/**
+		 * Regenerates the security code for the given key.
+		 *
+		 * @since 5.2.2
+		 *
+		 * @param string $key Key to generate security code.
+		 *
+		 * @return false|string Generated code or False if not generated.
+		 */
+		public function maybe_regenerate_security_code( $key ) {
+
+			$regenerate = apply_filters( 'tribe_tickets_regenerate_security_code_on_email_change', true );
```

Comment:
Needs hook doc

Reply:
Let's also pass $key into the filter because maybe there's some special logic someone might want to be able to utilize to make this determination.